src/App.tsx

```diff
@@ -121,23 +121,46 @@ function App(): ReactElement {
   // UTILITY METHODS /////////////////////////////////////////////////////////////
 
   /**
-   * Get a set of URL parameters that represent the current collection, dataset, feature, track,
-   * and frame information. (Convenience wrapper for `urlUtils.getUrlParams`.)
+   * Get a URL query string representing the current collection, dataset, feature, track,
+   * and frame information.
    */
   const getUrlParams = useCallback((): string => {
     let datasetParam: string | undefined = datasetKey;
-    if (!collection?.url) {
-      // The collection has no source file; use the dataset URL instead
+    let collectionParam = collection?.url;
+    // If there is no collection loaded (e.g. a single dataset URL was loaded),
+    // use the dataset URL instead.
+    if (collectionParam === null) {
       datasetParam = dataset?.manifestUrl;
+      collectionParam = undefined;
     }
-    return urlUtils.stateToUrlParamString({
-      collection: collection?.url,
+    // check if current color map range matches the default feature range; if so, don't include
+    // it in the URL parameter to reduce clutter.
+    let rangeIsDefault = false;
```

Comment:
seems like this could be a function on colorRamp class. Ie `rangeIsDefault = colorRamp.getRangeIsDefault(featureData, colorRampMin, colorRampMin`.

Also, could you just do a `if !dataset` return? at the top of this? hopefully to remove the multiple conditionals here.

Reply:
Ah, I think the comment here I had was a bit misleading. The range here isn't governed by the ColorRamp class, but is a min and max feature value for where the endpoints of the gradient (held by ColorRamp) should be applied. I've updated the comment accordingly.